lib/msf/core/handler/reverse_http.rb

```diff
@@ -325,20 +327,20 @@ def on_request(cli, req)
       request_summary = "#{conn_id} with UA '#{req.headers['User-Agent']}'"
 
       # Validate known UUIDs for all requests if IgnoreUnknownPayloads is set
-      if datastore['IgnoreUnknownPayloads'] && ! framework.db.get_payload({uuid: uuid.puid_hex})
-        print_status("Ignoring unknown UUID: #{request_summary}")
+      if datastore['IgnoreUnknownPayloads'] && ! framework.db.payloads({uuid: puid})
+        print_warning("Ignoring unknown UUID: #{request_summary}")
         info[:mode] = :unknown_uuid
       end
 
       # Validate known URLs for all session init requests if IgnoreUnknownPayloads is set
       if datastore['IgnoreUnknownPayloads'] && info[:mode].to_s =~ /^init_/
         payload_info = {
-            uuid: uuid.puid_hex,
+            uuid: puid,
         }
-        payload = framework.db.get_payload(payload_info)
+        payload = framework.db.payloads(payload_info).first
```

Comment:
Same as for `lib/msf/core/payload/uuid/options.rb`, this change had undesired impacts with using `workspaces`.